src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -2708,108 +2742,118 @@ protected virtual void MarkReflectionLikeDependencies (MethodBody body)
-									bindingFlags |= BindingFlags.Public;
+								if (parameters [0].ParameterType.MetadataType == MetadataType.String) {
+									ProcessActivatorCallWithStrings (i - 1, parameters.Count < 4);
 									break;
-								case Code.Ldc_I4_1 when parameters [1].ParameterType.MetadataType == MetadataType.Boolean:
-									parametersCount = 0;
+								}
+
+								var first_arg_instr = GetInstructionAtStackDepth (instructions, i - 1, methodCalledDefinition.Parameters.Count);
+								if (first_arg_instr < 0) {
+									ReportUnrecognizedReflectionCallPattern ($"Activator call '{methodCalled.FullName}' inside '{body.Method.FullName}' couldn't be decomposed");
 									break;
-								case Code.Ldc_I4_S when parameters [1].ParameterType.IsTypeOf ("System.Reflection", "BindingFlags"):
-									bindingFlags = (BindingFlags)(sbyte)second_argument.Operand;
+								}
+
+								if (parameters [0].ParameterType.IsTypeOf ("System", "Type")) {
+									declaringType = FindReflectionTypeForLookup (instructions, first_arg_instr + 1);
+									if (declaringType == null) {
+										ReportUnrecognizedReflectionCallPattern ($"Activator call '{methodCalled.FullName}' inside '{body.Method.FullName}' was detected with 1st argument expression which cannot be analyzed");
+										break;
+									}
+
+									BindingFlags bindingFlags = BindingFlags.Instance;
+									int? parametersCount = null;
+
+									if (methodCalledDefinition.Parameters.Count == 1) {
+										parametersCount = 0;
+									} else {
+										var second_arg_instr = GetInstructionAtStackDepth (instructions, i - 1, methodCalledDefinition.Parameters.Count - 1);
+										second_argument = instructions [second_arg_instr];
+										switch (second_argument.OpCode.Code) {
+											case Code.Ldc_I4_0 when parameters [1].ParameterType.MetadataType == MetadataType.Boolean:
+												parametersCount = 0;
+												bindingFlags |= BindingFlags.Public;
+												break;
+											case Code.Ldc_I4_1 when parameters [1].ParameterType.MetadataType == MetadataType.Boolean:
+												parametersCount = 0;
+												break;
+											case Code.Ldc_I4_S when parameters [1].ParameterType.IsTypeOf ("System.Reflection", "BindingFlags"):
+												bindingFlags = (BindingFlags)(sbyte)second_argument.Operand;
+												break;
+										}
+									}
+
+									MarkMethodsFromReflectionCall (declaringType, ".ctor", 0, bindingFlags, parametersCount);
 									break;
 								}
 							}
 
-							MarkMethodsFromReflectionCall (declaringType, ".ctor", 0, bindingFlags, parametersCount);
-							continue;
-						}
+							break;
+						//
+						// static T CreateInstance<T> ()
+						//
+						case "CreateInstance`1":
+							// Not sure it's worth implementing as we cannot expant T and simple cases can be rewritten
+							_context.LogMessage (MessageImportance.Low, $"Activator call '{methodCalled.FullName}' inside '{body.Method.FullName}' is not supported");
```

Comment:
Shouldn't this be reported as unrecognizable?

Reply:
Good catch.

Turns out the ``CreateInstance`1`` is not the correct name as reported by Cecil. Cecil reports generic methods without any modifications to their names - so in this case it's still `CreateInstance`.

My change fixes the problem and thus introduces a new unrecognized pattern in the linker. Which is technically a behavioral change. I think it's better to fix this though.